starter-code/scripts/views/articleView.js

```diff
@@ -37,6 +40,12 @@
   };
 
   // COMMENT: What does this method do?  What is it's execution path?
+  //This method ensures that multiple selections cannot be made before the selected
+  //information can be apended, and ensures that only one append can happen at a time.
+  //It takes the selected variable and uses the this property of the event listener
+  //grabs the id and removes the '-filter' and replaces it with that selected value.
+  //Whichever option (category or author) that was not selected is then emptied out so
+  //that it can be repopulated.
```

Comment:
Append is maybe not the right word to use here, but I think you have the write idea. This method is an event handler that only allows an event to happen the first time something is selected. Once a select event happens in our filters, we save a reference to the resource, whether it be an author or category (by targeting the css id selector and removing the suffix -filter [replacing it with an empty string, essentially nothing] so we're left with "author" or "category" -- this way we can deal with both filters in one method).  It then "selects" the title option from the **other** select box (as that title has a value of '') so that it reverts back to the original look, on the title text (as it wasn't being used to select the current filter).

Finally, we redirect our client to a route that we've laid out with Page.js, either /author/:authorName or /category/:categoryName, replacing any spaces with + signs in our URL.